src/librustdoc/clean/types.rs

```diff
@@ -759,53 +759,27 @@ impl Item {
         Some(tcx.visibility(def_id))
     }
 
-    fn attributes_without_repr(&self, tcx: TyCtxt<'_>, is_json: bool) -> Vec<String> {
-        const ALLOWED_ATTRIBUTES: &[Symbol] =
-            &[sym::export_name, sym::link_section, sym::no_mangle, sym::non_exhaustive];
+    /// Get a list of attributes excluding `#[repr]`.
+    ///
+    /// Only used by the HTML output-format.
+    fn attributes_without_repr(&self, tcx: TyCtxt<'_>) -> Vec<String> {
+        const ALLOWED_ATTRIBUTES: &[Symbol] = &[sym::non_exhaustive];
         self.attrs
             .other_attrs
             .iter()
             .filter_map(|attr| {
                 if let hir::Attribute::Parsed(AttributeKind::LinkSection { name, .. }) = attr {
                     Some(format!("#[link_section = \"{name}\"]"))
-                }
-                // NoMangle is special cased, as it appears in HTML output, and we want to show it in source form, not HIR printing.
-                // It is also used by cargo-semver-checks.
-                else if let hir::Attribute::Parsed(AttributeKind::NoMangle(..)) = attr {
+                } else if let hir::Attribute::Parsed(AttributeKind::NoMangle(..)) = attr {
```

Comment:
A bit of a drive-by, but should we update these to Rust 2024 edition form? `#[unsafe(no_mangle)], #[unsafe(export_name = ...)]` etc.

Reply:
Yeah, probably. We try to use latest edition syntax everywhere. Worth filling an issue if there's not one already.

Reply:
I'll see if I can open a PR and save us all a step or two 👀

Reply:
open here: https://github.com/rust-lang/rust/pull/143662